compiler/noirc_evaluator/src/ssa/ssa_gen/program.rs

```diff
@@ -98,3 +108,39 @@ impl Display for Ssa {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::ssa::ir::map::Id;
+
+    use crate::ssa::ssa_gen::Ssa;
+    use crate::ssa::{
+        function_builder::FunctionBuilder,
+        ir::{instruction::BinaryOp, types::Type},
+    };
+
+    #[test]
+    fn serialization_roundtrip() {
+        let main_id = Id::test_new(0);
+
+        // Compiling main
+        let mut builder = FunctionBuilder::new("main".into(), main_id);
+        let v0 = builder.add_parameter(Type::field());
+
+        let one = builder.field_constant(1u128);
+        let three = builder.field_constant(3u128);
+
+        let v1 = builder.insert_binary(v0, BinaryOp::Add, one);
+        let v2 = builder.insert_binary(v1, BinaryOp::Mul, three);
+        builder.terminate_with_return(vec![v2]);
+
+        let ssa = builder.finish();
+
+        let serialized_ssa = &serde_json::to_string(&ssa).unwrap();
+        let deserialized_ssa: Ssa = serde_json::from_str(serialized_ssa).unwrap();
+        let actual_string = format!("{}", deserialized_ssa);
+
+        let expected_string = "acir(inline) fn main f0 {\n  b0(v0: Field):\n    v3 = add v0, Field 1\n    v4 = mul v3, Field 3\n    return v4\n}\n";
```

Comment:
Would this be clearer if we made it a multiline or raw string? The indentation could look a bit off visually but I think the example would be easier to read.

Reply:
> Would this be clearer if we made it a multiline or raw string?

I was trying to use a raw string, but then it reads in all the extra spaces due to indentation and causes a test failure.

Reply:
I'll see about multiline

Reply:
Can't you omit the extra indentation then? The snippet may look like it is indented differently but it should be more readable, right?

Reply:
I pushed an update using multiline as it was easier than using raw strings.